File: Valid-Mountain-Array.py
class Solution(object):
    def validMountainArray(self, arr):
        """
        :type arr: List[int]
        :rtype: bool
        """
        i, j = 0, len(arr) - 1

        if len(arr) < 3:
            return False

        while i + 1 < len(arr) and arr[i] < arr[i + 1]:
            i += 1
        while j > 0 and arr[j] < arr[j - 1]:
            j -= 1

        return i > 0 and j < len(arr) - 1 and i == j

File: Valid-Mountain-Array.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& arr) {
        int i = 0, j = arr.size() - 1;

        if (arr.size() < 3) return false;

        while (i + 1 < arr.size() && arr[i] < arr[i + 1]) i++;
        while (j > 0 && arr[j] < arr[j - 1]) j--;

        return i > 0 && j < arr.size() - 1 && i == j;
    }
};
